### PR TITLE
Remove select.poll and improve subprocess

### PR DESCRIPTION
**Pull request by jstasiak**
@temoto Could you have a look at this? I'm not comfortable pushing to master directly as the subprocess module wasn't monkey patched at all until this change, there may be a reason for that.

This closes https://github.com/eventlet/eventlet/issues/293

### CODE REVIEW

**Comment by jstasiak** [comment]
Ignore the broken build, I forgot about few things, will fix soon.

File: eventlet/green/subprocess.py [hunk 1]
```diff
@@ -9,14 +9,15 @@
...
+__patched__ = ['call', 'check_call', 'check_open', 'Popen']
 to_patch = [('select', select), ('threading', threading), ('time', time)]
 
 if sys.version_info > (3, 4):
     from eventlet.green import selectors
     to_patch.append(('selectors', selectors))
 
 patcher.inject('subprocess', globals(), *to_patch)
-subprocess_orig = __import__("subprocess")
+subprocess_orig = patcher.original("subprocess")
```
Comment (temoto) [inline eventlet/green/subprocess.py]: 👍

**Comment by jstasiak** [comment]
After few iterations this is in a mergeable state now I believe.

**Comment by temoto** [comment]
Thank you very much.

**Pull request merged**